Malmo/src/MissionSpec.cpp

```diff
@@ -18,656 +18,625 @@
-            rrp.set(reward);
-        }
-        else
-        {
-            rrp->Marker().push_back(PointWithReward(x, y, z, amount, tolerance));
-        }
+        mission.put("Mission.AgentSection.AgentHandlers.RewardForReachingPosition.Marker.<xmlattr>.x", x);
+        mission.put("Mission.AgentSection.AgentHandlers.RewardForReachingPosition.Marker.<xmlattr>.y", y);
+        mission.put("Mission.AgentSection.AgentHandlers.RewardForReachingPosition.Marker.<xmlattr>.z", z);
+        mission.put("Mission.AgentSection.AgentHandlers.RewardForReachingPosition.Marker.<xmlattr>.reward", amount);
+        mission.put("Mission.AgentSection.AgentHandlers.RewardForReachingPosition.Marker.<xmlattr>.tolerance", tolerance);
     }
     
     void MissionSpec::observeRecentCommands()
     {
-        ObservationFromRecentCommands obs;
-        this->mission->AgentSection().front().AgentHandlers().ObservationFromRecentCommands( obs );
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromRecentCommands", "");
     }
     
     void MissionSpec::observeHotBar()
     {
-        ObservationFromHotBar obs;
-        this->mission->AgentSection().front().AgentHandlers().ObservationFromHotBar( obs );
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromHotBar", "");
     }
     
     void MissionSpec::observeFullInventory()
     {
-        ObservationFromFullInventory obs;
-        this->mission->AgentSection().front().AgentHandlers().ObservationFromFullInventory( obs );
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromFullInventory", "");
     }
     
     void MissionSpec::observeGrid(int x1,int y1,int z1,int x2,int y2,int z2,const std::string& name)
     {
-        AgentHandlers::ObservationFromGrid_optional& obs = this->mission->AgentSection().front().AgentHandlers().ObservationFromGrid();
-        if (!obs.present())
-        {
-            ObservationFromGrid observer;
-            observer.Grid().push_back(GridDefinition(Pos(x1, y1, z1), Pos(x2, y2, z2), name));
-            obs.set(observer);
-        }
-        else
-        {
-            obs->Grid().push_back(GridDefinition(Pos(x1, y1, z1), Pos(x2, y2, z2), name));
-        }
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromGrid.Grid.min.<xmlattr>.x", x1);
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromGrid.Grid.min.<xmlattr>.y", y1);
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromGrid.Grid.min.<xmlattr>.z", z1);
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromGrid.Grid.max.<xmlattr>.x", x2);
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromGrid.Grid.max.<xmlattr>.y", y2);
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromGrid.Grid.max.<xmlattr>.z", z2);
+        mission.put("Mission.AgentSection.AgentHandlers.ObservationFromGrid.Grid.<xmlattr>.name", name);
     }
```

Comment:
Another method that appends to an array - not sure if the replacement code is doing that or not?